### PR TITLE
[CARBONDATA-2198] Fixed bug for streaming data for bad_records_action as REDIRECT or IGNORE

### PR DESCRIPTION
**Pull request by geetikagupta16**
1. Refactored streaming functionality for bad_records_action as IGNORE or REDIRECT
2. Added related test cases

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/2736/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/3981/

**Comment by geetikagupta16** [comment]
@mohammadshahidkhan Please review. This issue is related to PR #1942

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/3737/

File: hadoop/src/main/java/org/apache/carbondata/hadoop/streaming/CarbonStreamRecordWriter.java [hunk 1]
```diff
@@ -181,75 +181,81 @@ private void initializeAtFirstRow() throws IOException, InterruptedException {
...
-        nullBitSet.set(i);
+    Object[] rowData = (Object[]) value;
+    currentRow.setRawData(rowData);
+    // parse and convert row
+    currentRow.setData(rowParser.parseRow(rowData));
+    CarbonRow updatedCarbonRow = converter.convert(currentRow);
+    if (updatedCarbonRow == null) {
+      output.skipRow();
+      currentRow.clearData();
+    } else {
```
Comment (mohammadshahidkhan) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/streaming/CarbonStreamRecordWriter.java]: @geetikagupta16 Thanks for working on this
Hope no change in the else part, only formatting changed due the if check.

Comment (geetikagupta16) [reply hadoop/src/main/java/org/apache/carbondata/hadoop/streaming/CarbonStreamRecordWriter.java]: @mohammadshahidkhan  Yes, else part remains the same

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3031/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4275/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3515/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4742/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4249/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/5172/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4456/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5017/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6007/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/4848/

**Comment by kunal642** [comment]
LGTM

**Comment by mohammadshahidkhan** [comment]
LGTM

**Pull request closed without merging**